### PR TITLE
[cudapoa] experimental commit to restructure files and remove void*

### LINKED ISSUE #407: [cudapoa] Replace the usage of `void*` with templated args
https://github.com/clara-genomics/ClaraGenomicsAnalysis/pull/397 introduces functions that take in `void *` arguments in the kernel host wrappers. A restructuring of the source files in cudapoa can potentially remove this requirement as this is requiring duplication of logic in some files (i.e. heuristics to choose between different sizer types).

example attempt in https://github.com/tijyojwad/ClaraGenomicsAnalysis/tree/jdaw/template-structure-changes

### PR DESCRIPTION
**Pull request by tijyojwad**
Closes #407

### CODE REVIEW

File: cudapoa/src/cudapoa_kernels.cuh [hunk 1]
```diff
@@ -8,33 +8,19 @@
...
-
-// Dimensions for Banded alignment score matrix
-#define WARP_SIZE 32
-#define CELLS_PER_THREAD 4
-#define CUDAPOA_BANDED_MATRIX_RIGHT_PADDING (CELLS_PER_THREAD * 2)
-
-#define CUDAPOA_THREADS_PER_BLOCK 64
-#define CUDAPOA_BANDED_THREADS_PER_BLOCK WARP_SIZE
-#define CUDAPOA_MAX_CONSENSUS_PER_BLOCK 512
+#include "cudapoa_kernels.cuh"
```
[the remark this answers is not in the document]
Comment (tijyojwad) [reply cudapoa/src/cudapoa_kernels.cuh]: Good catch! Removed

**Comment by tijyojwad** [comment]
rerun tests

**Comment by tijyojwad** [comment]
@r-mafi Realized some of the cmake changes were unnecessary for this PR, so removed them for now. Now mostly the cmake changes are in the `cudapoa` module, and slight updates to some others.

If you're good with this, let's merge it since it would be good to base the other PRs on top of this.

**Comment by r-mafi** [comment]
> If you're good with this, let's merge it since it would be good to base the other PRs on top of this.
@tijyojwad yes, looks good. thanks.

**Pull request merged**